### PR TITLE
Add json-ld contexts for agent registry

### PR DESCRIPTION
**Pull request by justinwb**
Adds JSON-LD contexts for:
- [Agent Registry](https://solid.github.io/data-interoperability-panel/specification/#ar-registry)
- Agent Registrations
  - [Social Agent Registration](https://solid.github.io/data-interoperability-panel/specification/#social-agent-registration)
  - [Application Registration](https://solid.github.io/data-interoperability-panel/specification/#application-registration)

For the agent registrations, there is a single context (agent-registration.jsonld), rather than a specific context for social agent and application registrations, respectively. Worth discussing whether we should have a context for each, but they use all the same properties, with the addition of `reciprocalRegistration` in the social agent registration. For now I stuck with the DRY approach of using one context for both, but open to thoughts / arguments to split.

### CODE REVIEW

**Review by acoburn — COMMENTED** [review]
I don't entirely understand why there is a `agentRegistryId` and `agentRegistrationId`, both of which map to `@id`. I would suggest omitting both of those and just including the following:

```
{
  "@context": {
      "id": "@id",
      "type": "@type",
      ...
  }
}
```

Otherwise, if both are used in a JSON document, the outcome may be unpredictable. (Unless, of course, there is a "very good reason" to do so)

File: proposals/specification/contexts/agent-registry.jsonld [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      },
+      "socialAgentRegistrations": {
+        "@id": "http://www.w3.org/ns/solid/interop#hasSocialAgentRegistration",
+        "@type": "@id",
+        "@container": [
+          "@id",
+          "@set"
+        ]
+      },
+      "applicationRegistrations": {
```
[the remark this answers is not in the document]
Comment (justinwb) [reply proposals/specification/contexts/agent-registry.jsonld]: Resolved in https://github.com/solid/data-interoperability-panel/pull/236/commits/e8cb1e455f39ee05c46ebc7dff7c29440bd49998

File: proposals/specification/contexts/agent-registration.jsonld [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      },
+      "registeredAgent": {
+        "@id": "http://www.w3.org/ns/solid/interop#registeredAgent",
+        "@type": "@id"
+      },
+      "reciprocalRegistration": {
+        "@id": "http://www.w3.org/ns/solid/interop#reciprocalRegistration",
+        "@type": "@id"
+      },
+      "accessGrant": {
```
[the remark this answers is not in the document]
Comment (justinwb) [reply proposals/specification/contexts/agent-registration.jsonld]: Resolved in https://github.com/solid/data-interoperability-panel/pull/236/commits/e8cb1e455f39ee05c46ebc7dff7c29440bd49998

**Comment by justinwb** [comment]
https://github.com/solid/data-interoperability-panel/pull/236/commits/e8cb1e455f39ee05c46ebc7dff7c29440bd49998 updates all contexts to consistently use "hasX" when the corresponding property uses "hasX" per @acoburn's review feedback

**Comment by justinwb** [comment]
> I don't entirely understand why there is a `agentRegistryId` and `agentRegistrationId`, both of which map to `@id`

@acoburn Yes this is good feedback. It was haphazardly done. We had a discussion at the end of the interop panel session today where we agreed on two things:

1. Revise the id approach (which I believe aligns with your suggestion above)
2. Use one context document rather than splitting up into many (would be interested to know whether you disagree with this choice)

**Comment by acoburn** [comment]
> Use one context document rather than splitting up into many (would be interested to know whether you disagree with this choice)

I would use a single document here.

I think of a context like a "domain". For example, your domain might be ornithology or activity data or photo libraries. One _could_ have a library of photos that track bird activity, but if _generally_ the domains are orthogonal, then keep the contexts orthogonal. There will always be a judgement call in terms of where to draw distinctions, but given the small size of these two JSON-LD contexts and the fact that they both relate to Solid Interop, I would be inclined to merge them into one.

The best argument for splitting them would be if one defined the "core" types for interop and the other was more of an example of an extension point with the expectation that other extension contexts might be used instead.

File: proposals/specification/interop.jsonld [hunk 3]
```diff
@@ -0,0 +1,106 @@
+{
+  "@context": {
+    "@version": 1.1,
+    "@protected": true,
+    "interop" : "http://www.w3.org/ns/solid/interop#",
+    "xsd" : "http://www.w3.org/2001/XMLSchema#",
+    "solid": "http://www.w3.org/ns/solid/terms#",
+    "id": "@id",
+    "type": "@id",
```
Comment (elf-pavlik) [inline proposals/specification/interop.jsonld:9]:
```suggestion
```

I don't think you even need `"type": "@type"`

**Review by elf-pavlik — APPROVED** [review]

**Pull request merged**